storage_service/common/management/commands/create_aip_replicas.py

```diff
@@ -0,0 +1,150 @@
+                self.error("No AIP Store location specified or set as default.")
+                sys.exit(1)
+
+        aips = Package.objects.filter(
+            current_location=aip_store_uuid, package_type="AIP", replicated_package=None
+        ).all()
+        if not aips:
+            self.error("No AIPs to replicate in location {}".format(aip_store_uuid))
+            sys.exit(1)
+
+        aips_count = len(aips)
+        success_count = 0
+        deleted_count = 0
+
+        self.info("AIPs to replicate: {}".format(aips_count))
+
+        for aip in aips:
+            if delete_existing_replicas:
+                self.info("Deleting existing replicas of AIP {}".format(aip.uuid))
+                aip_deleted_replicas_count = self._delete_replicas(aip.uuid)
+                deleted_count += aip_deleted_replicas_count
+
+            replicas_initial_count = get_replica_count(aip.uuid)
+
+            self.info("Creating new replicas for AIP {}".format(aip.uuid))
+            aip.create_replicas()
+
+            # Validate that new replicas were created.
+            replicas_count = get_replica_count(aip.uuid)
+            if not replicas_count > replicas_initial_count:
+                self.error("Replicas not created for AIP {}".format(aip.uuid))
+                continue
+
+            self.info("AIP {} successfully replicated".format(aip.uuid))
+            success_count += 1
+
+        self.success(
+            "Replica creation complete. {} existing replicas deleted. "
+            "New replicas created for {} of {} AIPs in location.".format(
+                deleted_count, success_count, aips_count
+            )
+        )
+
+    def _delete_replicas(self, aip_uuid):
+        """Delete all existing replicas of an AIP
+
+        :param aip_uuid: UUID of AIP whose replicas we are deleting.
+
+        :returns: Number of replicas deleted (int)
+        """
+        deleted_count = 0
+        existing_replicas = Package.objects.filter(replicated_package=aip_uuid).all()
+        for replica in existing_replicas:
+            try:
+                self._delete_replica(replica, aip_uuid)
+                deleted_count += 1
+            except ReplicaDeleteException as err:
+                self.error(
+                    "Unable to delete existing replica {} of AIP {}: {}".format(
+                        replica.uuid, aip_uuid, err
+                    )
+                )
+        return deleted_count
+
+    def _delete_replica(self, replica, aip_uuid):
+        """Delete replica from filesystem and database
+
+        :param replica: Package object of replicated AIP to delete.
+        :param aip_uuid: UUID of AIP whose replicas we are deleting.
+        """
+        # Delete from filesystem.
+        fs_deletion = replica.delete_from_storage()
+        fs_deletion_success = fs_deletion[0]
+        if not fs_deletion_success:
+            error_msg = fs_deletion[1]
+            raise ReplicaDeleteException(error_msg)
```

Comment:
I'm starting to think that we don't need custom exceptions when they don't have any specific behaviour/state or there is no specific control flow up in the stack. You could argue it's dead code after all and raising Exception would be just fine. What do you think?

Reply:
I don't have a strong opinion on this but suppose I'd slightly lean toward keeping the custom exception because it indicates that the exception was an anticipated one, and offers a bit of a hint to the end user about what happened. But I do see your point and would be fine with changing it if you prefer.